src/platform_impl/windows/icon.rs

```diff
@@ -160,3 +163,93 @@ pub fn unset_for_window(hwnd: HWND, icon_type: IconType) {
         SendMessageW(hwnd, WM_SETICON, icon_type as usize, 0);
     }
 }
+
+#[derive(Debug)]
+struct RaiiCursor {
+    handle: HCURSOR,
+}
+
+#[derive(Clone, Debug)]
+pub struct WinCursor {
+    inner: Arc<RaiiCursor>,
+}
+
+impl WinCursor {
+    pub fn as_raw_handle(&self) -> HICON {
+        self.inner.handle
+    }
+
+    fn from_handle(handle: HCURSOR) -> Self {
+        Self {
+            inner: Arc::new(RaiiCursor { handle }),
+        }
+    }
+
+    pub fn new(image: &CursorImage) -> Result<Self, io::Error> {
+        let mut bgra = image.rgba.clone();
+        bgra.chunks_exact_mut(4).for_each(|chunk| chunk.swap(0, 2));
+
+        let w = image.width as i32;
+        let h = image.height as i32;
+
+        let handle = unsafe {
+            let mask_bits: Vec<u8> = vec![0xff; ((((w + 15) >> 4) << 1) * h) as usize];
+            let hbm_mask = CreateBitmap(w, h, 1, 1, mask_bits.as_ptr() as *const _);
+            if hbm_mask == 0 {
+                return Err(io::Error::last_os_error());
+            }
+
+            let hdc_screen = GetDC(0);
+            if hdc_screen == 0 {
+                return Err(io::Error::last_os_error());
+            }
```

Comment:
Don't we have DC around on the window already for stuff like that?

Reply:
If we do I can't find it.